src/freenet/client/filter/M3UFilter.java

```diff
@@ -116,65 +117,63 @@ public void readFilter(InputStream input, OutputStream output, String charset, H
                     if (!isComment) {
                         // remove too long paths
                         if (fileIndex <= MAX_URI_LENGTH) {
-                            String uriold = new String(fileUri, 0, fileIndex, "UTF-8");
-                            // System.out.println(uriold);
-                            // clean up the URL: allow sub-m3us and mp3/ogg/flac (what we can filter)
-                            String filtered;
-                            try {
-                                String subMimetype = ContentFilter.mimeTypeForSrc(uriold);
-                                filtered = cb.processURI(uriold, subMimetype);
-                                // add prefix for the host name
-                                // for absolute path names,
-                                // because otherwise external
-                                // clients could be tricked into
-                                // accessing local files (and some
-                                // just don't work, especially not
-                                // with downloaded files). This
-                                // can however make downloaded
-                                // files leak information about
-                                // the local setup (host and
-                                // port).
+                            boolean lineIsEmpty = fileIndex == 0;
+                            if (!lineIsEmpty) {
```

Comment:
If `fileIndex` is `0` at this point it means that not a single byte was read for this line which means that the first character read was a New Line which also means we already aborted this loop in line 96.

Reply:
Except if the line was terminated by \r\n (I tried the cleaner version before).

Reply:
Oh, right. Carry on then!